benchmarks/benchmark.jl

```diff
@@ -59,3 +59,24 @@ u = fill!(similar(Jv), 1)
 Jtu = similar(x0)
 b_jtprod = @benchmark jtprod_residual!($rd_bamodel, $x0, $u, $Jtu)
 display(b_jtprod)
+
+nls = BundleAdjustmentModel(name)
+meta_nls = nls_meta(nls)
+
+function F!(Fx, x)
+  residual!(nls, x, Fx)
+end
```

Comment:
Shouldn't we do:
```
function F!(Fx, x, nls = nls)
  residual!(nls, x, Fx)
end
```
?
I haven't tested but I think it makes a difference allocation-wise.